crypto/schemes/enc/v1/README.md

```diff
@@ -0,0 +1,137 @@
+# Dapr encryption scheme v1: `dapr.io/enc/v1`
+
+This document contains the reference for the **Dapr encryption scheme v1**, identified by the header `dapr.io/enc/v1`.
+
+The Dapr encryption scheme is optimized for processing data as a stream. Data is chunked into multiple parts which are encrypted independently. This allows us to return data to callers as a stream, even when decrypting messages, being confident that we are not flushing unverified data to the client.
+
+> **Sources:** The encryption scheme that Dapr uses is heavily inspired by the [Tink wire format](https://developers.google.com/tink/wire-format) (from the Tink library maintained by Google), as well as by Filippo Valsorda's [age](https://age-encryption.org/v1), and Minio's [DARE](https://github.com/minio/sio).
+
+## Key
+
+Each message is encrypted with a 256-bit symmetric **File Key (FK)** that is randomly generated by Dapr for each new message. The key must be generated as 32 byte of output from a CSPRNG (such as Go's `crypto/rand.Reader`) and must not be reused for other files.
+
+The FK is wrapped using a key stored in a key vault (**Key Encryption Key (KEK)**) by Dapr. The result of the wrapping operation is the **Wrapped File Key (WFK)**. The algorithm used depends on the type of the KEK as well as the algorithms supported by the component:
+
+- For symmetric keys:
+    - AES-KW ([RFC 3394](https://www.rfc-editor.org/rfc/rfc3394.html)): `AES-KW`
+- For RSA keys:
+    - RSA OAEP with SHA-256: `RSA-OAEP-256`
+
+> Other key wrapping algorithms can be implemented in the future.
+
+## Ciphertext format
+
+The ciphertext is formatted as:
+
+```text
+header || binary_payload
+```
+
+## Header
+
+The **header** is human-readable and contains 3 items, each terminated by a line feed (`0x0A`) character:
+
+1. Name and version of the encryption scheme used. In this version of the spec, this is always `dapr.io/enc/v1`.
+2. The manifest, which is a JSON object.
```

Comment:
I think there is an implicit requirement that line feed characters (`0x0A`) are not part of the serialized JSON  output. [RFC 8259 - The JavaScript Object Notation (JSON) Data Interchange Format - Section 2](https://www.rfc-editor.org/rfc/rfc8259#section-2) mentions line feed as an allowed insignificant white space

> Insignificant whitespace is allowed before or after any of the six structural characters.

Something like

```json
{"k":"mykey","kw":1,"wfk":"hGYjwDpWEXEymSTFZ95zgX8krElb3Gqyls67R8zJA3k=","cph":1,"np":"Y3J5cHRvIQ==",
"this line break just before this key using a \u000A will break our parsing" :
"but it is valid JSON"}
``` 


```suggestion
2. The manifest, which is a JSON object with insignificant whitespace removed.
```

Reply:
```suggestion
2. The manifest, which is a JSON object, compacted with all the unnecessary whitespace removed.  
   Note that per the JSON specs, newline characters within a string are encoded as the string `\n`, which would not interpreted as newlines while scanning the manifest.
```